soroban-sdk/src/deploy.rs

```diff
@@ -53,61 +56,102 @@ impl Deployer {
         &self.env
     }
 
-    /// Get a deployer that deploys contracts that derive their contract IDs
-    /// from the current contract and the provided salt.
+    /// Get a deployer that deploys contract that derive the contract IDs
+    /// from the current contract and provided salt.
     pub fn with_current_contract(
         &self,
-        salt: &impl IntoVal<Env, BytesN<32>>,
-    ) -> DeployerWithCurrentContract {
-        let env = self.env();
-        DeployerWithCurrentContract {
-            env: env.clone(),
-            salt: salt.into_val(env),
+        salt: impl IntoVal<Env, BytesN<32>>,
+    ) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address: self.env.current_contract_address(),
+            salt: salt.into_val(&self.env),
         }
     }
 
-    #[doc(hidden)]
-    /// Get a deployer for contracts that derive their contract IDs from the
-    /// given contract ID and the provided salt.
-    pub fn with_other_contract(
+    /// Get a deployer that deploys contracts that derive the contract ID
+    /// from the provided address and salt.
+    ///
+    /// The deployer address must authorize all the deployments.
+    pub fn with_address(
+        &self,
+        address: Address,
+        salt: impl IntoVal<Env, BytesN<32>>,
+    ) -> DeployerWithAddress {
+        DeployerWithAddress {
+            env: self.env.clone(),
+            address,
+            salt: salt.into_val(&self.env),
+        }
+    }
+
+    /// Get a deployer that deploys an instance of Stellar Asset Contract
+    /// corresponding to the provided serialized asset.
+    ///
+    /// `serialized_asset` is the Stellar `Asset` XDR serialized to bytes. Refer
+    /// to `stellar-xdr` create for the exact definition.
```

Comment:
The type is in this crate, we can refer to it here.
```suggestion
    /// `serialized_asset` is the Stellar `Asset` XDR serialized to bytes. Refer
    /// to `[soroban_sdk::xdr::Asset]`
```